### PR TITLE
{CI} Auto merge release pull reqeust create by azclibot

### PR DESCRIPTION
**Pull request by wangzelin007**
---
image
We need to merge the PR created by azclibot in time.
Otherwise the index.json from the new PR does not contain the contents of the old PR.
And we must create a PR to fix the index.json conflict problem.
Such as: https://github.com/Azure/azure-cli-extensions/pull/4940 https://github.com/Azure/azure-cli-extensions/pull/4779

This PR will auto merge the PR created by azclibot.
For old steps, we still modify in Azure CLI Extensions Release pipeline.
image
For new steps, we add scripts in `scripts/release` directory. (For easy tracking and reviewing.)

There is a new pipeline `Azure CLI Extensions Release Test Auto Merge` to test auto merge.
image
image

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

File: scripts/release/auto_merge_pr.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        logger.debug(r)
+        logger.debug(r.text)
+        sys.exit(1)
+
+
+def main():
+    try:
+        # get azclibot token
+        with open('/tmp/token.txt') as f:
+            token = f.readline().replace("\n", "")
```
Comment (kairu-ms) [inline scripts/release/auto_merge_pr.py:45]: Why not fetch token from keyvault?

Comment (wangzelin007) [reply scripts/release/auto_merge_pr.py:45]: In old steps, we can use environment variable `Azure subscription` in Azure CLI option.
image
So it is easy to use azure cli command.

But new step I use python script which not support environment variable.
image
And after the job finished (about 4 mins) this agent will be delete, so I think it is safe.

Comment (jiasli) [reply scripts/release/auto_merge_pr.py:45]: Can't we use **Arguments**?

**Pull request closed without merging**